client/core/core.go

```diff
@@ -3373,13 +3377,44 @@ out:
 			if sinceLast >= 2*dc.tickInterval {
 				// The app likely just woke up from being suspended. Skip this
 				// tick to let DEX connections reconnect and resync matches.
+				// TODO: Need similar check in dc.tickAsset where block updates
+				// could trigger trade.tick().
 				log.Warnf("Long delay since previous trade check (just resumed?): %v. "+
 					"Skipping this check to allow reconnect.", sinceLast)
 				continue
 			}
 			updatedAssets := make(assetMap)
 			dc.tradeMtx.Lock()
 			for oid, trade := range dc.trades {
+				stuck, epochEnd := trade.isStuckAtEpochStatus()
+				if stuck {
+					// Orders stuck at epoch status are assumed to be inactive
+					// (Executed / Revoked) because staying at Epoch status for
+					// "so long" implies that we missed the preimage request for
+					// this trade or we sent the preimage but missed the match
+					// or nomatch request.
+					// TODO: Use the order_status route to determine and set the
+					// correct status for this trade. Batch orders of this nature
+					// in a single order_status request, and execute the request
+					// asynchronously to avoid delaying ticking other trades.
+					trade.mtx.Lock()
+					trade.returnCoins()
+					trade.mtx.Unlock()
+					delete(dc.trades, oid)
+					updatedAssets.count(trade.fromAssetID)
+					log.Infof("Order %v in epoch status with server time stamp %v, "+
+						"epoch end %v (%v ago) considered inactive.",
+						oid, trade.Prefix().ServerTime, epochEnd, time.Since(epochEnd))
+					continue
+				}
```

Comment:
I don't perform this check when orders are loaded from db to allow this tick cycle run the check and properly retire the order, including returning locked coins and updating the fromWallet balance.

`core.loadDBTrades` checks for and ignores inactive trades retrieved from db; I think we should only log a "didn't expect to load inactive trade from db" message there and allow the order to be retired in this tick cycle. That will allow core load the fundingCoins for the trade and return them later when the trade is retired. Especially since we no longer return coins on client shutdown, coins might be locked for the order that will remain locked.